### PR TITLE
Calculator

### PR DESCRIPTION
**Pull request by juandiego11116**

### CODE REVIEW

File: Calculator/Calculator/ViewController.swift [hunk 1]
```diff
@@ -18,65 +18,59 @@ class ViewController: UIViewController {
...
        
-       var aux = ""
-       var operationInArray: [String] = []
+        var aux: String = ""
+        var operationInArray: [String] = []
         
-        for char in mainLabel.text! {
-            if char == "0" || char == "1" || char == "2" || char == "3" || char == "4" || char == "5" || char == "6" || char == "7" || char == "8" || char == "9"{
+        for char in mainLabel.text ?? "0" {
+            if char.isNumber || char == "."{
```
Comment (RodriguesAndre) [inline Calculator/Calculator/ViewController.swift:25]: you're not handling localisation, is some countries `.` or `,` are used

File: Calculator/Calculator/ViewController.swift [hunk 2]
```diff
@@ -88,246 +82,50 @@ class ViewController: UIViewController {
...
-    
-    @IBAction func plus(_ sender: Any) {
-        if mainLabel.text == "0"{
-            
-            
-        } else if mainLabel.text?.last == "/" || mainLabel.text?.last == "*" || mainLabel.text?.last == "+" || mainLabel.text?.last == "-" {
-            
-        }else{
-            mainLabel.text?.append("+")
+        if characterSet.contains(String((mainLabel.text?.last)!)) {
```
Comment (RodriguesAndre) [inline Calculator/Calculator/ViewController.swift:109]: `!` can cause crash

File: Calculator/Calculator/ViewController.swift [hunk 3]
```diff
@@ -88,246 +82,50 @@ class ViewController: UIViewController {
...
+        } else {
+            mainLabel.text?.append(sender.titleLabel!.text ?? "0")
```
Comment (RodriguesAndre) [inline Calculator/Calculator/ViewController.swift:111]: `!` can cause crash

File: Calculator/Calculator/ViewController.swift [hunk 4]
```diff
@@ -88,246 +82,50 @@ class ViewController: UIViewController {
...
-        if mainLabel.text == "0"{
-            
-            
-        } else if mainLabel.text?.last == "/" || mainLabel.text?.last == "*" || mainLabel.text?.last == "+" || mainLabel.text?.last == "-" {
-            
-        }else{
-            mainLabel.text?.append("/")
-        }
+    @IBAction func operationsButton(_ sender: UIButton) {
+        let characterSet: String = "+-*/0"
```
Comment (RodriguesAndre) [inline Calculator/Calculator/ViewController.swift:107]: use better names, and here you are using a String and not NSCharacterSet which is mislleading

File: Calculator/Calculator/ViewController.swift [hunk 5]
```diff
@@ -88,246 +82,50 @@ class ViewController: UIViewController {
...
         }
         
+        let formatter = NumberFormatter()
         
-        let result = String(operationsArray[0]).components(separatedBy: ".")
-        if Int(result[1]) == 0 {
-            mainLabel.text = result[0]
-        }else{
+        if operationsArray[0].hasSuffix(".0") {
+            mainLabel.text = formatter.string(from: Double(operationsArray[0])! as NSNumber) ?? "n/a"
```
Comment (RodriguesAndre) [inline Calculator/Calculator/ViewController.swift:88]: ! can cause crash

File: Calculator/Calculator/ViewController.swift [hunk 6]
```diff
@@ -88,246 +82,50 @@ class ViewController: UIViewController {
...
-//            }
-//        }
     }
     
-    @IBAction func plusOrLessBotton(_ sender: Any) {
-        if mainLabel.text == "0" {
-            
-            
-        }else if mainLabel.text?.first == "-" {
+    @IBAction func addOrSubtractButton(_ sender: Any) {
```
Comment (RodriguesAndre) [inline Calculator/Calculator/ViewController.swift:94]: this name mean that you handle 2 types of operations here, which is not the case

File: Calculator/Calculator/ViewController.swift [hunk 7]
```diff
@@ -88,246 +82,50 @@ class ViewController: UIViewController {
...
+            mainLabel.text?.append(sender.titleLabel!.text ?? "0")
         }
     }
     
     @IBAction func ACBotton(_ sender: Any) {
         mainLabel.text = "0"
     }
     
+    @IBAction func numbers(_ sender: UIButton) {
+        mainLabel.text == "0" ? mainLabel.text = sender.titleLabel!.text : mainLabel.text?.append(sender.titleLabel!.text ?? "0")
```
Comment (RodriguesAndre) [inline Calculator/Calculator/ViewController.swift:120]: 👏  much smarter
But don't use `!`

File: Calculator/Calculator/ViewController.swift [hunk 8]
```diff
@@ -88,246 +82,50 @@ class ViewController: UIViewController {
...
+    }
+    
     @IBAction func pointDecimal(_ sender: Any) {
         if ((mainLabel.text?.contains(".")) == false){
             mainLabel.text?.append(".")
-        }else{
+        } else {
             mainLabel.text = mainLabel.text
```
Comment (RodriguesAndre) [inline Calculator/Calculator/ViewController.swift:127]: what's the point here?

File: Calculator/Calculator/ViewController.swift [hunk 9]
```diff
@@ -88,246 +82,50 @@ class ViewController: UIViewController {
...
-        }else{
-            mainLabel.text?.append("+")
+        if characterSet.contains(String((mainLabel.text?.last)!)) {
+        } else {
+            mainLabel.text?.append(sender.titleLabel!.text ?? "0")
         }
     }
     
     @IBAction func ACBotton(_ sender: Any) {
         mainLabel.text = "0"
```
Comment (RodriguesAndre) [inline Calculator/Calculator/ViewController.swift:116]: you should reset more things here

File: Calculator/Calculator/ViewController.swift [hunk 10]
```diff
@@ -88,246 +82,50 @@ class ViewController: UIViewController {
...
-            
-        }else{
-            mainLabel.text?.append("+")
+        if characterSet.contains(String((mainLabel.text?.last)!)) {
+        } else {
+            mainLabel.text?.append(sender.titleLabel!.text ?? "0")
         }
     }
     
     @IBAction func ACBotton(_ sender: Any) {
```
Comment (RodriguesAndre) [inline Calculator/Calculator/ViewController.swift:115]: wrong naming convention

**Comment by RodriguesAndre** [comment]
you are not using branch merge correctly. if you merge them, I can't check them

File: Calculator/Calculator/ViewController.swift (hunk 8 above)
Comment (juandiego11116) [reply Calculator/Calculator/ViewController.swift:127]: In another comment you say "wrong coding convention } else {", i thing i need to change it

**Pull request merged**